### PR TITLE
fix: comprehensive v3.0 audit — 122 TS errors, 17 test failures, stale docs, CI/CD

### PR DESCRIPTION
**Pull request by jbdevprimary**
## Summary

Audit of Kiro specs' "45/45 FULLY IMPLEMENTED" claim for Cognitive Dissonance v3.0 revealed significant gaps across TypeScript compilation, test infrastructure, documentation, CI/CD workflows, and spec accuracy. This PR addresses **all** findings.

- **TypeScript**: 122 → **0** compilation errors across 19 source files
- **Tests**: 80 passed / 17 failed → **129 passed / 0 failed** (12/12 suites)
- **Lint**: 25 errors → **0** Biome errors
- **Docs**: DESIGN.md, DESIGN_SYSTEM.md, .gemini/ prompts rewritten from v2 (Three.js) → v3.0 (Babylon.js 8)
- **CI/CD**: 5 workflow issues fixed (bundle calc, Maestro YAML, cd.yml token/artifact/EAS)
- **Kiro Specs**: Reclassified from "45/45 FULLY IMPLEMENTED" → "41/45 SUBSTANTIALLY IMPLEMENTED" with STUB/WORKAROUND annotations

### TypeScript Fixes (122 → 0 errors)
- Merge duplicate `update()` in DifficultyScalingSystem
- Add missing field declarations in DreamTypeHandler, HandPhysics
- Fix `IKeyboardEvent` incompatibility in KeyboardInputSystem
- Fix wrong singleton instantiation in SystemOrchestrator, ARSessionManager
- Add null guards in GamePhaseManager, ShatterSequence particle loops
- Remove unused fields, prefix unused stub params

### Test Fixes (80/97 → 129/129 passing)
- Fix Jest `transformIgnorePatterns` for pnpm nested `@babylonjs` paths
- Fix `spawnDreamFromSeed(number)` → `string` type mismatch in World.test.ts
- Fix singleton test access via `(Class as any).instance = null` pattern
- Add `noNaN: true` to all `fc.float()` PBT generators (IEEE 754 edge case)
- Fix method name mismatches, import paths, null safety across 12 test files
- Add `jest.mock()` for Babylon.js ESM subpath imports

### Documentation Updates
- Rewrite DESIGN.md from React+Three.js to Reactylon Native+Babylon.js 8
- Rewrite DESIGN_SYSTEM.md with tension-driven color system, GLSL shaders, PBR materials
- Update `.gemini/` prompt files for v3.0 stack references

### CI/CD Fixes
- Fix bundle size calculation: concatenate-then-gzip instead of per-file gzip
- Fix Maestro YAML `tapOn` syntax (nested map form)
- Fix cd.yml: GitHub Release `with.token`, web artifact retention, EAS Build token guard

### Kiro Specs Accuracy
- Reclassify to 41/45 with 4 deferred items (tasks 25, 38, 39, 42)
- Add STUB/WORKAROUND annotations to affected subtasks
- Change tasks 27, 30 parents from `[X]` to `[~]` (partial completion)

## Test plan
- [x] `npx tsc --noEmit` — 0 errors
- [x] `pnpm test` — 129/129 tests passing, 12/12 suites
- [x] `pnpm lint` — 0 errors (76 warnings, all in `__tests__/` for `any` casts)
- [x] CI/CD YAML files validated
- [x] Maestro YAML files validated
- [ ] Verify CI pipeline passes on push

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: e2e/web/smoke.spec.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    await page.goto('/');
+    await page.waitForTimeout(5000);
+    
+    // Filter out known non-critical errors
+    const criticalErrors = errors.filter(
+      (error) => !error.includes('DevTools') && !error.includes('favicon')
+    );
+    
+    expect(criticalErrors).toHaveLength(0);
+  });
```
[the remark this answers is not in the document]
Comment (jbdevprimary) [reply e2e/web/smoke.spec.ts]: Acknowledged — valid suggestion but out of scope for this audit PR (TS errors, test failures, stale docs). Tracking for follow-up.

File: src/accessibility/DiegeticAccessibility.ts [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+  onVoiceCommand(command: string): void {
+    const upperCommand = command.toUpperCase().trim();
+
+    // Check if command is a valid keycap letter
+    if (this.validKeys.includes(upperCommand)) {
+      this.stabilizeKey(upperCommand);
+    } else {
+      console.log(`DiegeticAccessibility: Unrecognized command "${command}"`);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (jbdevprimary) [reply src/accessibility/DiegeticAccessibility.ts]: Acknowledged — valid suggestion but out of scope for this audit PR (TS errors, test failures, stale docs). Tracking for follow-up.

**Comment by jbdevprimary** [comment]
Closing: this branch represents a complete architectural overhaul (Expo SDK 55, Metro, 31 systems, Maestro E2E) that diverges fundamentally from main's working Next.js game. The 251 files / 25k lines of changes aren't mergeable — this would be a full rewrite. Archiving the branch for reference; future v3.0 work should start from current main.
